apps/api/src/domain/services/query-services/song-query.service.ts

```diff
@@ -13,12 +17,46 @@ export class SongQueryService extends ResourceQueryService<Song, ISongViewModel>
 
     buildViewModel(
         song: Song,
-        { resources: { audioItem: allAudioItems, mediaItem: allMediaItems } }: InMemorySnapshot
+        {
+            resources: { audioItem: allAudioItems, mediaItem: allMediaItems },
+            contributor: allContributors,
+        }: InMemorySnapshot
     ): ISongViewModel {
-        return new SongViewModel(song, allAudioItems, allMediaItems);
+        return new SongViewModel(song, allAudioItems, allMediaItems, allContributors);
     }
 
     getDomainModelCtors(): DomainModelCtor<BaseDomainModel>[] {
         return [Song];
     }
+
+    protected override async fetchRequiredExternalState(): Promise<InMemorySnapshot> {
+        const tagSearchResult = await this.repositoryProvider.getTagRepository().fetchMany();
```

Comment:
We should use a Promise.all to optimize these queries. Note that when we move to event-sourced views, the joins will be done eagerly.

Reply:
Added.